### ISSUE TITLE
Resource monitor creation fails, yet objects are created.

### ISSUE THREAD

**Issue by rasmus-kristensen** [Snowflake-Labs/terraform-provider-snowflake#1754]
**Provider Version**
0.62.0

**Terraform Version**

1.4.5

**Describe the bug**

When attempting to create new resource monitor in snowflake the "terraform init" runs successfully and as expected however the apply fails, yet the objects are still created in snowflake.
When attempting to terraform apply again it now complains that the objects already exists and are not found in the state.

![image](https://user-images.githubusercontent.com/106582326/234255330-c5d265b0-3baa-4405-906d-867fc588bda2.png)

```
│ Error: Plugin did not respond
│ 
│   with module.snowflake_monitor["MRCH"].snowflake_resource_monitor.monitor,
│   on resource-monitor/main.tf line 17, in resource "snowflake_resource_monitor" "monitor":
│   17: resource "snowflake_resource_monitor" "monitor" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-snowflake_v0.62.0 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 37 [running]:
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.ReadResourceMonitor(0x140008b6380, {0x102069960?, 0x140008b8b60})
        github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/resource_monitor.go:322 +0x9cc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x102094140?, {0x102094140?, 0x1400084f560?}, 0xd?, {0x102069960?, 0x140008b8b60?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:712 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x140005bb880, {0x102094140, 0x1400084f560}, 0x140009e6410, {0x102069960, 0x140008b8b60})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:1015 +0x468
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400078eb40, {0x102094140?, 0x1400084f440?}, 0x140004aa340)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:613 +0x400
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140001b10e0, {0x102094140?, 0x1400084ec90?}, 0x1400067e4e0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x102030900?, 0x140001b10e0}, {0x102094140, 0x1400084ec90}, 0x14000162000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140007b2000, {0x10209a580, 0x14000502340}, 0x14000c98fc0, 0x140007ba990, 0x1029158d0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xb7c
google.golang.org/grpc.(*Server).handleStream(0x140007b2000, {0x10209a580, 0x14000502340}, 0x14000c98fc0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x290

Error: The terraform-provider-snowflake_v0.62.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Expected behavior**
terraform apply should not fails as it does actually create the objects, in version 0.61.0 this is not an issue.

**Code samples and commands**

```
resource "time_offset" "next_month" {
  offset_months = 1
}

resource "snowflake_resource_monitor" "monitor" {
  name                    = "MRCH_MONITOR"
  credit_quota            = 100
  frequency               = "MONTHLY"
  start_timestamp         = "${formatdate("YYYY-MM", time_offset.next_month.rfc3339)}-01 00:00:00"
  suspend_trigger         = 110
  warehouses              = values(merge(local.warehouses...))
}
```

**Additional context**

**Comment by tamathew** [comment]
I have yet another issue with resource monitor. 

- It always detects a change to datetime format and tries to apply it and when you apply it, it fails saying :
"Must specify frequency and start time together"

Timestamp changes -> 
start_timestamp :
"2023-05-03T00:00:00Z"
change to
"2023-05-03 00:00 PST"

**Comment by sfc-gh-jcieslak** [comment]
Yeah, I've this a lot in other issues. I handled it and the timestamps shouldn't cause those issues anymore in the upcoming version. I'll also test it with this particular case, Thank You.

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋
The new and refactored resource monitor was released yesterday in version 0.96.0 of the provider. Please migrate with [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0950--v0960) and let me know if the issue has been resolved so we could close the ticket. Thanks in advance 🙏.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity. Please, create another issue if you think the problem is still not resolved.
